### ISSUE TITLE
MssqlSession.query is not escaping double quote correctly

### ISSUE THREAD

**Issue by dalee-bis** [inspec/inspec#4392]
# Describe the problem

When `MssqlSession.query` is escaping the `"` character, it is doing so by prefixing it with a backslash (e.g. `\"`). This does not escape the quote character. It should be escaped by adding an additional double quote character (e.g. `""`).

Unfortunately, due to the current implementation, it is impossible to work around the issue (any use of `"` will add a backslash before so it cannot be doubled up.

Example SQL Query: `PRINT '"Test!"';`

Inspec Usage: `mssql_session.query("PRINT '\"Test!\"';")`

Generated command: `sqlcmd -Q "set nocount on; PRINT '\"Test\"!';" -W -w 1024 -s ','`

Output:
```
Msg 105, Level 15, State 1, Server EC2AMAZ-D3S9TNL, Line 1
Unclosed quotation mark after the character string ''.
```

Correct command: `sqlcmd -Q "set nocount on; PRINT '""Test""!';" -W -w 1024 -s ','`

Output:
```
"Test"!
```

## Possible Solution

Change https://github.com/inspec/inspec/blob/a35d37043e65ce5fd528d58cd48ecb369c74a432/lib/inspec/resources/mssql_session.rb#L56 to:

`escaped_query = q.gsub(/\\/, '\\\\').gsub(/"/, '""').gsub(/\$/, '\\$')`

**Comment by zenspider** [comment]
This is also true of oracledb_session.rb and probably others. We should do a sweep of all the db oriented resources.

**Comment by aaronlippold** [comment]
https://github.com/inspec/inspec/pull/1939

**Comment by kekaichinose** [comment]
Closed in #4393
